dist/api_data.py

```diff
@@ -186,66 +186,85 @@ def __str__(self):
+                            the number of triggers for which to wait''',
+                            min='1', max='4294967295'),
+                ]),
+            ConnectionControlPoint('Thread 1', 'Wait for trigger', 'Always', '', r'''
+                Thread 2 waits for thread 1 to get here.''',
+                type='category', subconfig= [
+                    # Action configuration parameters
+                    Config('wait_count', '1', r'''
+                            the number of triggers for which to wait''',
+                            min='1', max='4294967295'),
+                ]),
+            ConnectionControlPoint('Thread 2', 'Wait for trigger', 'Always', '', r'''
+                Thread 3 waits for thread 2 to get here.''',
+                type='category', subconfig= [
+                    # Action configuration parameters
+                    Config('wait_count', '1', r'''
+                            the number of triggers for which to wait''',
+                            min='1', max='4294967295'),
+                ]),
+            ConnectionControlPoint('Thread 3', 'Wait for trigger', 'Always', '', r'''
+                Thread 4 waits for thread 3 to get here.''',
+                type='category', subconfig= [
+                    # Action configuration parameters
+                    Config('wait_count', '1', r'''
+                            the number of triggers for which to wait''',
+                            min='1', max='4294967295'),
+                ]),
+            ConnectionControlPoint('Thread 4', 'Wait for trigger', 'Always', '', r'''
+                Thread 5 waits for thread 4 to get here.''',
+                type='category', subconfig= [
+                    # Action configuration parameters
+                    Config('wait_count', '1', r'''
+                            the number of triggers for which to wait''',
+                            min='1', max='4294967295'),
+                ]),
+        ]),
+    Config('per_session_control_points', '', r'''
+        Configure concurrent determinism through session control points''',
+        type='category', subconfig= [
+            SessionControlPoint('Thread 0', 'Sleep', 'Always',
+                '', r'''
+                Thread 0 performs a sleep on the session.''',
+                type='category', subconfig= [
+                    # Action configuration parameters
+                    Config('seconds', '3', r'''
+                            the number of seconds for which to wait''',
+                            min='1', max='4294967295'),
+                    Config('microseconds', '0', r'''
+                        the number of microseconds for which to wait''',
+                        min='1', max='4294967295'),
+                    Config('skip_count', '1', r'''
+                        the number of skips until we sleep''',
+                        min='1', max='4294967295'),
+                ]),
+        ])
```

Comment:
Move lines 244-262 to the replacement for old line 246 given below.

Reply:
I have made this change now.